zio-http/jvm/src/main/scala/zio/http/netty/server/ServerInboundHandler.scala

```diff
@@ -87,12 +87,17 @@ private[zio] final case class ServerInboundHandler(
             )
             releaseRequest()
           } else {
-            val req  = makeZioRequest(ctx, jReq)
-            val exit = handler(req)
-            if (attemptImmediateWrite(ctx, req.method, exit)) {
+            val req = makeZioRequest(ctx, jReq)
+            if (!validateHostHeader(req)) {
```

Comment:
@kyri-petrou I am not sure we should check this here. I think it is overhead and only if we parse it to the typed version we should check it, I think.

Reply:
I think the overhead should be relatively very small, although according to [this](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Host#:~:text=A%20400%20Bad%20Request%20status%20code%20may%20be%20sent%20to%20any%20HTTP/1.1%20request%20message%20that%20lacks%20or%20contains%20more%20than%20one%20Host%20header%20field.) the server _MAY_ respond with 400 if the host header is missing or is invalid.

To be honest I think a better approach would be to add a flag to the config `validateHeaders` (with a default to `false`) which will:
1. set [this](https://github.com/zio/zio-http/blob/22caa5a7920603955a500848edb4fc0ce892c575/zio-http/jvm/src/main/scala/zio/http/netty/server/ServerChannelInitializer.scala#L68) to true
2. enable any custom header validations

Reply:
Also, ideally we want to keep all header validation happening in one place. I think the only way we can do this is by extending the default encoder and adding any additional validation within it.

https://github.com/zio/zio-http/blob/22caa5a7920603955a500848edb4fc0ce892c575/zio-http/jvm/src/main/scala/zio/http/netty/server/ServerChannelInitializer.scala#L63

Reply:
Also, ideally we want to keep all header validation happening in one place. I think the only way we can do this is by extending the default encoder and adding any additional validation within it.

https://github.com/zio/zio-http/blob/22caa5a7920603955a500848edb4fc0ce892c575/zio-http/jvm/src/main/scala/zio/http/netty/server/ServerChannelInitializer.scala#L63